main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import "fmt"
+
+func worker(id int, jobs <-chan int, results chan<- int) {
+	for j := range jobs {
+		go func() {
+			switch j % 3 {
+			case 0:
+				j = j * 1
+			case 1:
+				j = j * 2
+				results <- j * 2
+			case 2:
+				results <- j * 3
+				j = j * 3
+			}
+		}()
+	}
+}
+
+func main() {
+	jobs := make(chan int)
+	results := make(chan int)
+	for i := 1; i <= 1000000000; i++ {
```

Comment:
same comment as above, use a variable to contain the loop i value every time that the go method runs

Reply:
also as good practice, we can use a variable instead of a literal for the loop size like 
```
maxCounter := 1000000
for i := 0; i <= maxCounter; i++ {
...
```

just for readibility and maintenance purposes